lib/datasets.go

```diff
@@ -161,12 +116,24 @@ func (r *DatasetRequests) Save(p *SaveParams, res *repo.DatasetRef) (err error)
 		return fmt.Errorf("option to make dataset private not yet implimented, refer to https://github.com/qri-io/qri/issues/291 for updates")
 	}
 
-	ds, body, secrets, err := actions.UpdateDataset(r.node, p.Dataset)
```

Comment:
The bulk of the work that needs to be checked here is improving the logic performed by `actions.NewDataset` and `actions.UpdateDataset`, with the [rfc](https://github.com/qri-io/rfcs/blob/master/text/0017-define_dataset_creation.md#steps) as the source of truth. 

I think a solid starting point would be to declare the new core package and move:
* `actions.NewDataset` -> `core.PrepareDatasetNew`
* `actions.UpdateDataset` -> `core.PrepareDatasetUpdate`. 
The `action` func that saves a dataset should then check for a `ds.Prev` value, and call one of those two core functions based on the presence of `ds.Prev`. This is kinda laid out in the [rfc reference section](https://github.com/qri-io/rfcs/blob/master/text/0017-define_dataset_creation.md#reference-level-explanation), but feel free to consider it a loose guide. For example, I think we can get away without `actions.prepareDataset`.

The logic in `NewDataset` and `UpdateDataset` is awful, but I think we'll have an easier time if we keep those calls & have this PR accomplish the task of merging new & save. If we get those methods moved into the new core package we can then do the work of cleaning them up in subsequent PRs.

These changes are a good start. The failing API tests that show missing/altered `bodyPath` and `path` values are an indication that logic from prior `actions.NewDataset` and `actions.UpdateDataset` are missing.

Reply:
Speaking in person, @b5 and I decided to hold off on the new `core` functions for a later PR. I added comments describing the intention for these new functions.

Failing API tests were mostly a consequence of the /new API route not being deleted yet.